Ship analytics for Homebrew.

mikemcquaid opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran `brew tests` with your changes locally?

I believe Homebrew's Analytics are ready to :ship:. This makes them opt-out by default. They will be accompanied by a tweet and mailing list post announcing this to people and pointing them towards the 
https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Analytics.md#analytics documentation.

@Homebrew/maintainers If there's any **blockers** in releasing analytics to everyone (i.e. we **cannot** ship analytics until this is fixed): please let me know in this issue. Feel free to suggest "nice to haves" but please be explicit about whether a comment is a suggestion/idea for something in the future or you consider it a hard blocker on releasing this feature.

I'll leave this open for at least a few days to solicit feedback.

Thanks!

--

Commit message:
For more information and opt-out instructions please read:
https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Analytics.md#analytics

xu-cheng commented on Library/Homebrew/utils/analytics.rb:
```diff
@@ -8,7 +8,6 @@ def analytics_label
 
 def report_analytics(type, metadata = {})
-  return unless ENV["HOMEBREW_ANALYTICS"]
   return if ENV["HOMEBREW_NO_ANALYTICS"]
```
We should add this env in manpage.

mikemcquaid replied:
:+1:

mikemcquaid replied:
Done in https://github.com/Homebrew/brew/pull/79

mikemcquaid commented:
@Homebrew/maintainers It sounds like there's no more objections here so I'll be :ship:ing this tomorrow or this weekend if I don't hear otherwise.

mikemcquaid commented:
@josegonzalez asked for a "Where" and "Who" section where the latter could be a maintainer document that details how to get access (e.g. ask someone already with access and list them).

apjanke commented:
Sorry for the lateness of the response here; I had to do some reading up on this since I'm not familiar with web analytics.

My only concern is that "impossible" in [the documentation](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Analytics.md#analytics) is pretty strong language when we're making assertions about privacy protection in terms of what Homebrew developers and Google can do with the data. Maybe we should downgrade that to a "can't", or report what Google themselves say the identifiability limitations are.

This is a suggestion, not a blocker.

mikemcquaid commented:
@apjanke Thanks. I'll change Google's "impossible" to something less strong but leave Homebrew's as-is.

Other @Homebrew/maintainers have suggested we announce this a week in advance so I'll do that when I consider this PR ready to :ship:

mikemcquaid commented:
I've addressed all feedback so this is ready to :ship:. I'm going to announce this now to the Homebrew community through Twitter and the mailing list and will merge this pull request on or after Saturday 23rd April at 0800 UTC.

mikemcquaid commented:
Relevant tweet: https://twitter.com/MacHomebrew/status/721290118703751168
Relevant mailing list post: https://groups.google.com/forum/#!topic/homebrew-discuss/7Xdx9NnNmt0

mikemcquaid commented:
Reminder that I'm 🚢ing this as-is tomorrow.

ilovezfs commented:
:cake: :balloon: :tada:
